### PR TITLE
QTA-138-Added prop alwaysShowAddToCart  show add to cart button quick order

### PR DESCRIPTION
**Pull request by jailsonacct**
…invalid SKU

#### What does this PR do? \*
An alwaysShowAddToCart property was added when validating whether or not to show the add to cart button. By declaring quickorder-upload in store-theme as true, the button will be shown even if some Sku is invalid. By clicking on the add to cart button, the valid ones will be added and the invalid ones will be ignored. If the property does not exist, validation continues as it was: The button will only be shown if all Sku's are valid.

#### How to test it? \*

Access the workspace  https://qta38--lojaobramax.myvtex.com/quickorder  and add a worksheet with SKUS, valid and invalid.
Click on the add to cart button, which will now appear. Valid items must be added to the cart.

SKU's list
[prods_with_invalids.xlsx](https://github.com/vtex-apps/quickorder/files/12186702/prods_with_invalids.xlsx)

### Before
[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2023.07.27-15_32_49.webm](https://github.com/vtex-apps/quickorder/assets/116591771/126269a5-7ee6-4750-872d-9ad177ec72a2)

### After
[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2023.07.27-15_30_48.webm](https://github.com/vtex-apps/quickorder/assets/116591771/09e150cb-e0ab-4d5e-b864-35e5bd33912c)

### CODE REVIEW

File: react/UploadBlock.tsx [hunk 1]
```diff
@@ -115,16 +116,16 @@ const UploadBlock: FunctionComponent<
 
   const onReviewItems = (items: any) => {
```
Comment (carlosananias) [inline react/UploadBlock.tsx:117]: @jailsonacct que tal otimizarmos o código com a função some() que retorna um true?

```
const onReviewItems = (items: any) => {
  const { alwaysShowAddToCart, state, setState } = this;

  const show = !items.some((item: any) => item.error);

  setState({
    ...state,
    reviewItems: items,
    reviewState: true,
    showAddToCart: alwaysShowAddToCart || show,
    textAreaValue: GetText(items),
  });
};
```

**Review by carlosananias — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,17 +7,21 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Removed
```
[the remark this answers is not in the document]
Comment (jailsonacct) [reply CHANGELOG.md]: @carlosananias Já tinha mudado para Added devido a adição da prop

**Review by Rudge — APPROVED** [review]

**Comment by Rudge** [comment]
@jailsonacct The action lint ran after approval and detected some errors. Could you fix them for merge, please?

**Comment by jailsonacct** [comment]
@Rudge What fixes? I saw that you changed yarn.lock there, which had not been changed before.
image

**Comment by Rudge** [comment]
> @Rudge What fixes? I saw that you changed yarn.lock there, which had not been changed before. I removed the change in yarn.lock again. image

@jailsonacct I executed the `npx eslint --ext js,jsx,ts,tsx . --fix` and maybe it updated the yarn.lock too. It isn't a problem to update this file.

**Review by Rudge — APPROVED** [review]

**Pull request merged**